### DISCUSSION TITLE
A sustainable solution to GUI scaling issues

### DISCUSSION THREAD

**Discussion by lucas-wilkins** [SasView/sasview#2429] [General]
The GUI has various issues around scaling on different systems, and for this reason we added preferences that allows users to choose how they scale things. However, it's not a particularly sustainable approach, and it's still very hard to get everything looking right on every system.

PyQt6 should deal with scaling differently, so do we just wait to see what it is like after we move?

Are there other strategies that we could employ to enable a more consistent look and feel?

**Comment by butlerpd** [comment]
Are we moving to PyQt6 first or PySide first? I thought I heard rumblings at the code camp and before that folks were looking to move to PySide (the BSD compatible python wrapping of Qt)?  That said a better solution to the gui scaling issues would be very welcome I think.  The current solution does not appear to me that sustainable either.

**Comment by rozyczko** [comment]
We are moving first to PySide2 then updating PySide2 to PySide6. I want to do this internally in stages, but also don't want to merge anything until the PySide6 version is properly working. 
I am waiting now for the 5.0.6 release so I can continue based off of that code base.

**Reply by lucas-wilkins** [reply]
> I am waiting now for the 5.0.6 release so I can continue based off of that code base.

I'm not sure basing changes off a codebase that has deviated from main is the way forward.

**Reply by butlerpd** [reply]
good point.  5.0.6 is 99% cherry picked from main so main already has that codebase. The only deviation is for patches to things that have been fixed differently in main already so will not be coming back. Any bug fixes to RC1 should of course be propagated to main as well.... unless I misunderstood your point @rozyczko?
